components/BugsnagTask.brs

```diff
@@ -243,7 +243,17 @@ function createDevicePayload()
 		device["deviceId"] = deviceInfo.GetChannelClientId()
 	end if
 
-	device["firmwareVersion"] = deviceInfo.GetOSVersion()
+	if FindMemberFunction(deviceInfo, "GetOSVersion") <> invalid
+		device["firmwareVersion"] = deviceInfo.GetOSVersion()
+	else 
+		version = deviceInfo.GetVersion()
+		device["firmwareVersion"] = {
+			major: Val(version.mid(2, 1))
+			minor: Val(version.mid(4, 2))
```

Comment:
The only problem I see here is that we're resolving `40` here as the minor version where it should be `4`. Not sure if adding additional logic to resolve the correct minor version is worth it, thoughts @MilosRasic? This is going to be correct once the devices start upgrading to FW version 10, but right now we're reporting a minor version as 40.

Reply:
Is this just constraining it to one characters? In that case, no additional logical, just do `minor: Val(version.mid(4, 1))` instead and problem solved :)

If not, then what's the logic? I'm not sure I understand tbh. Maybe add a comment here with sample string returned by `deviceInfo.GetVersion()` so that someone reading this in the future can have easier time understanding what we're doing here.

Reply:
Here's an example of what I'm talking about:

![Screen Shot 2021-03-15 at 5 28 42 PM](https://user-images.githubusercontent.com/25148427/111187049-e99db880-85b3-11eb-92eb-74918257053f.png)

`version` is just regular `GetOSVersion()` which we were using before, concatenated into string.

`_version` is return value of `GetVersion()`, and `__version` is string built with:

```brs
_version = deviceInfo.GetVersion()
__version = version.mid(2, 1) + " " + version.mid(4, 2) + " " + version.mid(8, 4)
```

Reply:
@Puritanic are you sure that `version` is same as `GetOSVersion`? I see no mention of this in the [documentation](https://developer.roku.com/en-ca/docs/references/brightscript/interfaces/ifdeviceinfo.md#getversion-as-string), especially since they mention that `GetVersion` returns **only** major, minor and build number. No mention of the revision number. 

@MilosRasic on the same link, you can find their example that minor can be `08`, so I'd say we will need 2 characters

Reply:
Sorry, I haven't left code impl for the `version` variable, it's build like this:
```brs
osVersion = deviceInfo.GetOSVersion()
version = osVersion.major + "." + osVersion.minor + "." + osVersion.revision + "." + osVersion.build
```

Reply:
But, would it be correct? I know this is not valid semver, but given their example `034.08E01185A` it translates to major 4, minor 08 and build 1185. Maybe I'm following this blindly, or misunderstanding something.

For versions higher than 9.1 this will not be an issue since we'd be using `GetOSVersion` which returns everything correctly.

All this aside, how would you prefer this to be handled, so I can fix this ASAP?

Reply:
Well, given that `deviceInfo.GetOSVersion()` formatted result is `9.4.0.4200` and `deviceInfo.GetVersion()` formatted result is `9.40.n/a.4200` I would say that it's safe to assume that `0` in `40` is a revision number if we compare these two results. Thoughts @MilosRasic?

Reply:
Yeah, let's just consider it `${majorVersion}.${minorVersion}${patchVersion}`. Even if we get it wrong, there's no way we'll get same version numbers with `deviceInfo.GetOSVersion()` on one device and `deviceInfo.GetVersion()` on another.

If usage in production reveals any problems with this, we can fix them later, when we have better info.

Reply:
@MilosRasic are you saying that we're good with the current implementation @iperzic proposed?

Reply:
No, I think in his example, it should be equivalent to 3.0.8, and you screenshot would be 9.4.0

Since we don't report "patch" version the 8 in 3.0.8 is lost and we'd report
major: 3
minor: 0
build: 1185

sound okayish?

Reply:
I will convert the third digit to patch number 👍

Reply:
Refactored in eca834be